autoload/neomake/makers/ft/go.vim

```diff
@@ -47,3 +48,17 @@ function! neomake#makers#ft#go#govet()
             \ '%-G%.%#'
         \ }
 endfunction
+
+function! neomake#makers#ft#go#gometalinter()
+    " Only run a subset of gometalinter for speed, users can override with:
+    " let g:let g:neomake_go_gometalinter_args = ['--disable-all', ....]
+    "
+    " All linters are only warnings, the go compiler will report errors
+    return {
+        \ 'args': ['--disable-all', '--enable=gosimple', '--enable=staticcheck'],
+        \ 'append_file': 0,
+        \ 'cwd': '%:h',
+        \ 'mapexpr': 'neomake_bufdir . "/" . v:val',
+        \ 'errorformat': '%W%f:%l:%c:%m',
```

Comment:
ah, just what i said above, it lets vim know whether it's an error or a warning. Example:
before:
```
version.go|16 error| syntax error: non-declaration statement outside function body
version.go|6 col 2 warning| warning: "fmt" imported but not used (errcheck)
version.go|6 col 2 warning| warning: "fmt" imported but not used (errcheck)
version.go|16 col 1 warning| warning: expected declaration, found 'IDENT' wat (golint)
version.go|16 col 1 warning| error: expected declaration, found 'IDENT' wat (vet)
version.go|16 col 1 warning| warning: expected declaration, found 'IDENT' wat (errcheck)
```

after:
```
version.go|16 error| syntax error: non-declaration statement outside function body
version.go|6 col 2 warning| "fmt" imported but not used (errcheck)
version.go|6 col 2 warning| "fmt" imported but not used (errcheck)
version.go|16 col 1 warning| expected declaration, found 'IDENT' wat (golint)
version.go|16 col 1 error| expected declaration, found 'IDENT' wat (vet)
version.go|16 col 1 warning| expected declaration, found 'IDENT' wat (errcheck)
```

Differences to note:
1) gometalinter's duplicated severity prefix ("error:" or "warning:") is stripped out of the error message, leaving only vim's error/warning category (which can be syntax-styled as usual)
2) vim's error/warning designation is now correct on line 5 (gometalinter reports it as an error, but the previous efm marks everything as a warning because that's the default error type (`%t`); after my change, vim and gometalinter agree)